packages/cozy-scanner/src/DocumentQualification.jsx

```diff
@@ -0,0 +1,201 @@
+  constructor(props) {
+    super(props)
+    const { categoryLabel = null, itemId = null } = props.initialSelected || {}
+    this.defaultFilename = `Scan_${new Date().toISOString().replace(/:/g, '-')}`
+    this.state = {
+      selected: { categoryLabel, itemId },
+      filename: this.defaultFilename,
+      hasUserWrittenFileName: false
+    }
+    this.textInput = React.createRef()
+  }
+
+  getFilenameFromCategory = (item, t) => {
+    const realItem = getItemById(item.itemId)
+    if (realItem) {
+      const name = t(`Scan.items.${realItem.label}`)
+      return `${name.replace(/ /g, '-')}_${new Date()
+        .toLocaleDateString()
+        .replace(/:/g, '-')
+        .replace(/\//g, '-')}`
+    } else {
+      return this.defaultFilename
+    }
+  }
+
+  onSelect = item => {
+    const { t, editFileName } = this.props
+    const { hasUserWrittenFileName } = this.state
+    let filename = null
+    if (!hasUserWrittenFileName) {
+      filename = this.getFilenameFromCategory(item, t)
+    } else {
+      filename = this.state.filename
+    }
+    this.setState({ selected: item })
+    this.handleFileNameChange(filename)
+
+    const { onQualified } = this.props
+    if (onQualified) {
+      const realItem = getItemById(item.itemId)
+      /* ATM, we only accept JPG extension from the scanner. So 
+      we hardcode the filename extension here. 
+
+      If we can't editFileName, then we don't need to use the 
+      generated filename 
+      */
+      if (editFileName) {
+        onQualified(
+          realItem ? realItem : undefined,
+          filename + filename_extension
+        )
+      } else {
+        onQualified(realItem ? realItem : undefined)
+      }
+    }
+  }
+  /**
+   * Method used to synchronize our internal state and
+   * our parent state if needed
+   */
+  handleFileNameChange = filename => {
+    const { onFileNameChanged } = this.props
+    this.setState({ filename })
+    onFileNameChanged && onFileNameChanged(filename + filename_extension)
+  }
+
+  render() {
+    const { t, title, editFileName } = this.props
+    const { selected, filename, hasUserWrittenFileName } = this.state
+    return (
+      <MuiCozyTheme>
+        {editFileName && (
+          <>
+            <Label htmlFor={id_filename_input}>{t('Scan.filename')}</Label>
+            <InputGroup
+              fullwidth
+              append={<span className="u-pr-1">{filename_extension}</span>}
+              className="u-bdrs-3"
+            >
+              <Input
+                placeholder={t('Scan.filename')}
+                value={filename}
+                onChange={event => {
+                  //If the user write something once, we don't want to rename the file automatically anymore
+                  if (!hasUserWrittenFileName) {
+                    this.setState({ hasUserWrittenFileName: true })
+                  }
+                  //If we left an empty value, then we reset the behavior
```

Comment:
```suggestion
                  // If we left an empty value, then we reset the behavior
```

Reply:
we should have this rule in our eslint / prettier config if you want one space after `//`.